apps/web/src/pages/EmployeeProfile/components/CareerDevelopmentSection/CareerDevelopmentSection.tsx

```diff
@@ -0,0 +1,503 @@
+        fr
+        localized
+      }
+    }
+    execCoachingInterest {
+      value
+      label {
+        en
+        fr
+        localized
+      }
+    }
+  }
+`);
+
+const UpdateEmployeeProfileCareerDevelopment_Mutation = graphql(/* GraphQL */ `
+  mutation UpdateEmployeeProfileCareerDevelopment(
+    $id: UUID!
+    $employeeProfile: UpdateEmployeeProfileInput!
+  ) {
+    updateEmployeeProfile(id: $id, employeeProfile: $employeeProfile) {
+      userPublicProfile {
+        id
+      }
+    }
+  }
+`);
+
+export type FormValues = Pick<
+  UpdateEmployeeProfileInput,
+  | "organizationTypeInterest"
+  | "moveInterest"
+  | "mentorshipInterest"
+  | "execCoachingInterest"
+> & {
+  mentorshipStatus?: string | null;
+  execInterest?: "yes" | "no";
+  execCoachingStatus?: string | null;
+};
+
+interface CareerDevelopmentSectionProps {
+  employeeProfileQuery: FragmentType<
+    typeof EmployeeProfileCareerDevelopment_Fragment
+  >;
+  careerDevelopmentOptionsQuery: FragmentType<
+    typeof EmployeeProfileCareerDevelopmentOptions_Fragment
+  >;
+}
+
+const CareerDevelopmentSection = ({
+  employeeProfileQuery,
+  careerDevelopmentOptionsQuery,
+}: CareerDevelopmentSectionProps) => {
+  const intl = useIntl();
+  const { userAuthInfo } = useAuthorization();
+  const [{ fetching }, executeMutation] = useMutation(
+    UpdateEmployeeProfileCareerDevelopment_Mutation,
+  );
+
+  const employeeProfile = getFragment(
+    EmployeeProfileCareerDevelopment_Fragment,
+    employeeProfileQuery,
+  );
+
+  const careerDevelopmentOptions = getFragment(
+    EmployeeProfileCareerDevelopmentOptions_Fragment,
+    careerDevelopmentOptionsQuery,
+  );
+
+  const labels = getLabels(intl);
+  const isNull = hasAllEmptyFields(employeeProfile);
+  const { isEditing, setIsEditing, icon } = useToggleSectionInfo({
+    isNull,
+    emptyRequired: false,
+    fallbackIcon: QuestionMarkCircleIcon,
+    optional: true,
+  });
+
+  const handleError = () => {
+    toast.error(
+      intl.formatMessage({
+        defaultMessage: "Failed updating career development preferences",
+        id: "CKxLR/",
+        description:
+          "Message displayed when a user fails to update the employee profile information",
+      }),
+    );
+  };
+
+  const dataToFormValues = ({
+    organizationTypeInterest,
+    moveInterest,
+    mentorshipStatus,
+    mentorshipInterest,
+    execInterest,
+    execCoachingStatus,
+    execCoachingInterest,
+  }: EmployeeProfile): FormValues => ({
+    organizationTypeInterest:
+      organizationTypeInterest?.map((x) => x.value) ?? [],
+    moveInterest: moveInterest?.map((x) => x.value) ?? [],
+    mentorshipStatus: mentorshipStatusToFormValues(mentorshipStatus),
+    mentorshipInterest: mentorshipInterest?.map((x) => x.value) ?? [],
+    execInterest: boolToYesNo(execInterest),
+    execCoachingStatus: execCoachingStatusToFormValues(execCoachingStatus),
+    execCoachingInterest: execCoachingInterest?.map((x) => x.value) ?? [],
+  });
```

Comment:
Could this be defined outside of the component so it doesn't need to be recreated on every mount?

Reply:
Moved in [22c8483](https://github.com/GCTC-NTGC/gc-digital-talent/pull/12680/commits/22c8483a97b8f6cf19dacc81c622f83c80124ce0)